k8s/reconcilers/deployment.go

```diff
@@ -439,6 +450,7 @@ func (r *DeploymentReconciler) updateCorrelationIdMetaData(ctx context.Context,
 	correlationId := uuid.New()
 	r.patchOperationStartTime(object, operationStartTimeKey)
 	object.GetAnnotations()[constants.AzureCorrelationIdKey] = correlationId.String()
+	object.GetAnnotations()[constants.PreviousAzureCorrelationIdKey] = correlationId.String()
```

Comment:
object.GetAnnotations()[constants.AzureCorrelationIdKey] = object.GetAnnotations()[constants.PreviousAzureCorrelationIdKey]
object.GetAnnotations()[constants.PreviousAzureCorrelationIdKey] = correlationId.String()

Reply:
Thanks for @FireDefend 's comments. I think this is the right way to distinguish whether the CR update is from cloud service or locally. Usually such as Azure, it only update correlation id in annotation. But PreviousAzureCorrelationIdKey is a little weird, it doesn't record previous correlation id.

Maybe it is better to change it as AzureCorrelationId and RunningCorrelationId.

Reply:
Thanks for the suggestion, has fixed